lib/travis/listener/schemas.rb

```diff
@@ -123,32 +138,44 @@ def self.event_details(event_type, payload)
             head:       payload['pull_request']['head']['sha'][0..6],
             ref:        payload['pull_request']['head']['ref'],
             user:       payload['pull_request']['head']['user']['login'],
-            sender:     payload['sender']['login']
+            sender:     parse_sender_from(payload)
           }
         when 'push'
           {
             repository: payload["repository"]["full_name"],
             ref:        payload['ref'],
             head:       payload['head_commit'] && payload['head_commit']['id'][0..6],
-            commits:   (payload["commits"] || []).map {|c| c['id'][0..6]}.join(","),
-            sender:     payload['sender']['login']
+            commits:    (payload["commits"] || []).map {|c| c['id'][0..6]}.join(","),
+            sender:     parse_sender_from(payload)
           }
         when 'create', 'delete', 'repository', 'check_run', 'check_suite'
           {
             action:     payload['action'],
             repository: payload["repository"]["full_name"],
-            sender:     payload['sender']['login']
+            sender:     parse_sender_from(payload)
           }
         when 'installation', 'installation_repositories'
           {
             action:       payload['action'],
             installation: payload["installation"]["account"]["login"],
-            sender:       payload['sender']['login']
+            sender:       parse_sender_from(payload)
           }
         else
           { }
         end
       end
+
+      # Some payloads come in that are missing a `sender` field for one reason
+      #   or another, but they do seem to have a `pusher` field, which has a
+      #   `name` field that is the same as the ['sender']['login'] value.
+      #
+      def self.parse_sender_from(payload)
```

Comment:
don't we use the user id from the payload?

Reply:
We do downstream, but in Listener we're only using the `decoded_payload` for logging; for downstream processing (next stop: Gatekeeper!) we are passing along the raw `params[:payload]` value.

That said, this /may/ just push the error down into Gatekeeper (if indeed it IS an error).. What's the best way to connect with :octocat: to get more info or guidance on this? We /might/ categorizing a non-push event as a push event..